test/e2e/user_management_test.go

```diff
@@ -377,8 +377,9 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 				mur.Status.ProvisionedTime.String())
 
 			// The user should be set to deactivating, but not deactivated
-			userSignup, err = hostAwait.WaitForUserSignup(t, userSignup.Name, wait.UntilUserSignupHasConditions(
-				wait.ConditionSet(wait.Default(), wait.ApprovedAutomatically(), wait.Deactivating())...))
+			userSignup, err = hostAwait.WaitForUserSignup(t, userSignup.Name, wait.UntilUserSignupHasScheduledDeactivationTime(),
+				wait.UntilUserSignupHasConditions(
+					wait.ConditionSet(wait.Default(), wait.ApprovedAutomatically(), wait.Deactivating())...))
```

Comment:
you combine `wait.Default()` together with `wait.Deactivating()`
the default has 
```go
{
	Type:   toolchainv1alpha1.UserSignupUserDeactivatedNotificationCreated,
	Status: corev1.ConditionFalse,
	Reason: "UserIsActive",
},
{
	Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
	Status: corev1.ConditionFalse,
	Reason: "UserNotInPreDeactivation",
},
```
and the deactivating:
```go
{
	Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
	Status: corev1.ConditionTrue,
	Reason: "NotificationCRCreated",
},
```
TBH, this is a bit confusing

Reply:
ah, I missed that, sorry. 
Anyway, this is pretty confusing - why do we combine the `wait.Default()` together with `wait.Deactivating()` when they expect different values for the same condition

Reply:
I'm not sure, but we can discuss it outside of this PR.